### PR TITLE
Update supproted platforms

### PR DESCRIPTION
**Pull request by JohnRoesler**
Signed-off-by: John Roesler <johnrroesler@gmail.com>

### Description

Clarifies the platforms that are actually tested in this cookbook.

We can't test freebsd as there isn't a dokken image. Any thoughts?

Dropping fedora as specifically supported.

### Contribution Check List

- [x] All tests pass.
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -57,12 +52,17 @@ platforms:
     intermediate_instructions:
       - RUN /usr/bin/apt-get update
 
-- name: opensuse-leap
+- name: opensuse-leap-42
```
Comment (damacus) [inline .kitchen.dokken.yml]: I think the latest is now -15

File: .kitchen.yml [hunk 2]
```diff
@@ -17,13 +17,13 @@ platforms:
 - name: amazon-linux
   driver_config:
     box: mvbcoding/awslinux
+- name: amazonlinux-2
+  driver_config:
+    box: ywatase/amzn2
```
Comment (damacus) [inline .kitchen.yml]: I like this, however I wonder if we should build our own regularly with boxcutter/bento?

**Comment by damacus** [comment]
You'll want to update the kitchen-orb to 1.0.1, as it removes fedora support :)

File: .kitchen.yml (hunk 2 above)
Comment (JohnRoesler) [reply .kitchen.yml]: Sure, could you point me in the right direction on this? Maybe open an issue and I'll tackle in a separate PR / could update it then in other cookbooks that are using amazon.

Comment (damacus) [reply .kitchen.yml]: https://github.com/chef/bento

Comment (damacus) [reply .kitchen.yml]: And here's the box we need to copy https://github.com/ywatase/packer-amazon-linux2

Comment (damacus) [reply .kitchen.yml]: (I'd like to be using sous-chefs for both tbh)

Comment (JohnRoesler) [reply .kitchen.yml]: Tim has a PR in flight for this it seems https://github.com/chef/bento/pull/1144

**Comment by JohnRoesler** [comment]
@damacus any insight into the failures with amazonlinux 2? https://circleci.com/gh/sous-chefs/haproxy/1162

**Comment by damacus** [comment]
I think we're missing headers
```
 STDERR: src/haproxy.c:66:10: fatal error: systemd/sd-daemon.h: No such file or directory
     #include <systemd/sd-daemon.h>
              ^~~~~~~~~~~~~~~~~~~~~
```
Have we classified amazon Linux 2 as a systemd platform?

**Comment by damacus** [comment]
That, platform family section looks like we should extract it out into a helper now 🤔

**Comment by JohnRoesler** [comment]
> That, platform family section looks like we should extract it out into a helper now 🤔

Created an issue for a separate PR to that #340

**Pull request merged**